dotcom-rendering/scripts/webpack/gu-stats-report-plugin.js

```diff
@@ -0,0 +1,158 @@
+const fetch = require('node-fetch');
+const os = require('os');
+const { exec } = require('child_process');
+
+class GuStatsReportPlugin {
+	constructor(config) {
+		this.buildName = config?.buildName;
+		this.project = config?.project;
+		this.team = config?.team;
+		this.buildCount = 0;
+		this.gitBranch = null;
```

Comment:
`this.gitBranch = this.fetchGitBranch()` could be easier to test as the fetch function would return the value for testing.

Reply:
The reason I couldn't do this (though I would have liked to) is because the calls for git info are asynchronous, so they cannot be put in a class constructor.

Defining these as either `null`, `undefined` or a default value in a constructor therefor makes it easier to read what properties will be set in the class even if they can't be assigned their value in the constructor (I forget if you can assign properties to a call in JS that haven't be initialised in the constructor, but I still think it's good practice).

There is then an inherent contract that these calls will complete before the initial webpack build, while I wouldn't expect them to take much more than a few hundred milliseconds at the most, it would be worth keeping in mind.

The alternative is we wrap these functions in Promises, and call them as part of the stats report function. However with up to 3 builds running simultaneously, it seems unnecessary to call them for every build & re-build.